UnitsNet/CustomCode/UnitAbbreviationsCache.cs

```diff
@@ -15,43 +16,47 @@ namespace UnitsNet
         ///     if no abbreviations are found with a given culture.
         /// </summary>
         /// <example>
-        ///     User wants to call <see cref="UnitParser.Parse{TUnitType}" /> or <see cref="Length.ToString()" /> with Russian
-        ///     culture, but no translation is defined, so we return the US English definition as a last resort. If it's not
+        ///     User wants to call <see cref="UnitParser.Parse{TUnitType}(string,IFormatProvider?)" /> or <see cref="Length.ToString()" /> with Russian
+        ///     culture, but no translation is defined, so we return the US English (en-US)  definition as a last resort. If it's not
         ///     defined there either, an exception is thrown.
         /// </example>
         internal static readonly CultureInfo FallbackCulture = CultureInfo.InvariantCulture;
 
         /// <summary>
-        ///     The default singleton instance with the default configured unit abbreviations, used for ToString() and parsing of quantities and units.
+        ///     The static instance used internally for ToString() and Parse() of quantities and units.
         /// </summary>
-        /// <remarks>
-        ///     Convenience shortcut for <see cref="UnitsNetSetup"/>.<see cref="UnitsNetSetup.Default"/>.<see cref="UnitsNetSetup.UnitAbbreviations"/>.<br />
-        ///     You can add custom unit abbreviations at runtime, and this will affect all usages globally in the application.
-        /// </remarks>
         public static UnitAbbreviationsCache Default => UnitsNetSetup.Default.UnitAbbreviations;
 
-        private QuantityInfoLookup QuantityInfoLookup { get; }
+        /// <summary>
+        ///     Gets the lookup table for quantity information used by this cache.
+        /// </summary>
+        /// <remarks>
+        ///     This property provides access to the <see cref="QuantityInfoLookup" /> instance that contains
+        ///     information about quantities and their associated units. It is used internally to map units
+        ///     to their abbreviations and vice versa.
+        /// </remarks>
+        internal QuantityInfoLookup Quantities { get; }
 
         /// <summary>
         /// Culture name to abbreviations. To add a custom default abbreviation, add to the beginning of the list.
         /// </summary>
         private ConcurrentDictionary<AbbreviationMapKey, IReadOnlyList<string>> AbbreviationsMap { get; } = new();
 
         /// <summary>
-        ///     Create an empty instance of the cache, with no default abbreviations loaded.
+        ///      Create an instance of the cache and load all the built-in quantities defined in the library.
         /// </summary>
+        /// <returns>Instance for mapping any of the built-in units.</returns>
         public UnitAbbreviationsCache()
-            : this(new QuantityInfoLookup([]))
+            :this(UnitsNetSetup.Default.QuantityInfoLookup)
```

Comment:
I kind of liked how the previous default ctor created an empty instance, and in order to create a new instance with defaults loaded you'd have to call `CreateDefault()`. Most people will use the singleton instance anyway, but it just seems more conventional to me.

Reply:
The goal here is to have the list of `QuantityInfo`s provided via the constructor, unlike before where a new `UnitInfo` was created when trying to get / map an abbreviation to a `TUnit` that doesn't exist in the `QuantityInfoLookup`.

The natural meaning IMO is that this creates an empty cache (meaning nothing is cached) using the default quantities (for mapping). In the 🐲  version I've got a few more constructor overloads, which should make it easier to construct / customize the default list.

Reply:
Okay maybe I just don't understand it, to me it seems like we're getting a cache with all built-in unit abbreviations when calling `new UnitAbbreviationsCache()` ?

Reply:
Yes, but it's an _empty cache_, as one would expect from a call to `new Cache()`

Reply:
Of course it has always been an empty cache, it's just that before it was an empty cache _and_ an empty list of quantities to map (which was problematic).

Reply:
Right, and this is more a theoretical exercise to challenge the design, but it seems nice to be able to use the pieces on their own without involving all the built-in unit definitions.

```cs
// For some reason, we don't want to initialize with HowMuch at this point
var cache = new UnitAbbreviationsCache();

// Instead add it later, in one of two ways
cache.AddQuantity(HowMuch.Info);
cache.Quantities.Add(HowMuch.Info);

// Then map the abbreviations
cache.MapUnitToDefaultAbbreviation(HowMuchUnit.Some, "sum");
```

Reply:
I guess it comes down to that it feels more intuitive to get this behavior from the default ctor, than having to call `CreateEmpty()`.

Reply:
Oh, that would be very difficult to accomplish.. Unless you want to add a message bus or something 😄. Of course we could have an assumption that the user would call `AddQuantity` before attempting to `Map` or `Get` it but again- that would make it impossible to use the `FrozenDictionary`, not to mention the concurrency issues once again, please let's not.. (_insert meme here_).

The `UnitParser` is now really lightweight, the user should be able to create one on the fly, but for the initial mapping (of the `DefaultAbrreviations`) I really hope that we could use something like `UnitsNet.Configure(c=> ... add custom stuff here)` at the app start.

Reply:
Alright, this is not an important point for me. I get that the new design forces you to either use the default ctor, with built-in quantities, or pass your own quantities, and that it becomes immutable at that point. That's fine.

We'll still need to accomodate adding custom quantities at runtime for the default singleton though, but this should be doable either by calling something early on startup as you say or swapping the immutable instance with a new one.